docs/DeveloperGuide.adoc

```diff
@@ -238,6 +239,93 @@ with the app.
 
 ==== Change Background and Font Color
 
+// tag::export[]
+=== [Proposed] Export Resume feature
+The Export Resume feature supports two main actions: previewing the content of a `Resume` item, and
+then exporting it as a .pdf file.
+
+==== rpreview: preview a `Resume`
+===== Proposed Implementation
+`rpreview` is supported by the new `Command`, `ResumePreviewCommand` and the additional method `Resume#getPreview()`
+which return the content of the `Resume` in textual format.
+
+Given below is an example usage scenario:
+
+Step 1. The user launches ResuMé. After loading data from storage to the application memory, the list of resumes in the
+ResumeBook could either contain some resumes, or is empty.
+
+Step 2. The user executes `rpreview INDEX`. If the specified resume index is out of bound, ResuMé throws an error message.
+
+Step 3. The application retrieves the correct `Resume` item and call `getPreview()` on that resume.
+
+The following activity diagram summarises what happens when a user executes `rpreview` command:
+
+image::ExportPreviewActivityDiagram.png[]
+
+
+===== Design Considerations
+===== Aspect: Where `rpreview` is displayed
+
+* *Alternative 1 (current choice):* Display in the same panel as `view`
+** Pros: No significant change to UI component.
+** Cons: Multiple commands needed if user finds out about a typo in an item, wants to view the item in details,
+fix the typo and then check the preview again to ensure there is no more error.
+
+* *Alternative 2:* Create a separate UI panel to display the resume preview. Additionally, this panel could be implemented
+such that it automatically updates when the content of the `Resume` item is edited.
+** Pros: User can see the resume preview and the item detail panels at the same time, hence saving time switching between views.
+** Cons: Too many panels could be confusing for the user to navigate. The space is also often wasted since user does not
+need to use `rpreview` regularly.
+
+==== rgen: generate a .pdf file from a `Resume`
+===== Proposed Implementation
+This feature utilises the external *iText* library. When using `rgen`, the user could specify the desired name of
+the generated file, which will be saved in the root directory of the project.
+
+`rgen` implements the following operations:
+
+* `create(String)` and `create()` - initialises the .pdf file, either with the input `String` name specified by the user,
+or with the same name as the `Resume` name if the file name is not specified.
+* `addSection(String)` - starts a new section with the `String` name (e.g. "Internship").
+* `populateSection(UniqueItemList)` and `insertItem(Item)` - populates the section with items in the same order as they
+are in the `UniqueItemList`. When an `Item` is inserted, it is correctly formatted depending on the item type.
+
+Given below is an example usage scenario:
+
+Step 1. The user launches ResuMé. After loading data from storage to the application memory, the list of resumes in the
```

Comment:
Is the accented e in ResuMé intended?

Reply:
Hahahah kind of cuz usually resumé is written with that accent thingy. But we can discuss the official naming some time later 😸